Add date-only mode to DateTimePickerControl

mattsherman opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR adds date-only support to DateTimePickerControl, through the `isDateOnlyPicker` prop. This mode will be needed for the new product management experience, where scheduled sale to and from fields are date only.

Needed for #34538

### How to test the changes in this Pull Request:

1. Run unit tests: `pnpm --filter=@woocommerce/components run test -- src/date-time-picker-control`
2. Interact with Storybook stories and make sure DateTimePickerControl works as expected, based on the `isDateOnlyPicker` prop: `pnpm --filter=@woocommerce/storybook storybook`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

mattsherman committed:
Remove border from top of date picker

mattsherman committed:
Add isDateOnlyPicker, rename is12HourPicker

mattsherman committed:
Add tests for isDateOnlyPicker

mattsherman committed:
Add date only story

mattsherman committed:
Changelog

louwie17 commented on packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx:
```diff
@@ -75,20 +77,36 @@ export const DateTimePickerControl: React.FC< DateTimePickerControlProps > = ( {
 		null
 	);
 
+	const displayFormat = useMemo( () => {
```
I think the use of `useMemo` here is a bit overkill as we are not doing any expensive parsing, and we are returning a string (so we don't have to worry about new object references).
It might be worth just removing the use of it.

mattsherman replied:
Good point, I'll remove that when I rebase and put it up for re-review!

louwie17 reviewed (dismissed):
This tested well and code look "mostly" good haha I did leave one suggestion in the code, but it's an optional suggestion, which is why I did approve it.

louwie17 commented:
Also @mattsherman you may want to do a rebase, that should fix the E2E tests I think

mattsherman committed:
Remove unnecessary useMemo

mattsherman commented:
@louwie17 I removed the `useMemo` and rebased against trunk. This could use another approval when you get the chance. Hopefully the E2E tests are cleared up by the rebase.

mattsherman commented:
@louwie17 Hold off on that re-review... I need to rebase again to fix conflicts now that I've merged #35077

mattsherman committed:
Fix tests use of is12HourPicker

mattsherman committed:
Fix up issues missed from rebase conflicts

louwie17 reviewed (approved):
LGTM 🚀 thanks for the rebase and removing `useMemo`
